spanner/quickstart.rb

```diff
@@ -32,11 +32,8 @@
 database_client = spanner.client instance_id, database_id
 
 # Execute a simple SQL statement
-database_client.transaction do |transaction|
-  results = transaction.execute "SELECT 1"
-
-  results.rows.each do |row|
-    puts row
-  end
+results = database_client.execute_query "SELECT 1"
```

Comment:
@olavloite This will not use single use transaction. To use single use transaction need to provide `single_use` argument.

i.e 
```
database_client.execute_query "SELECT 1", single_use: { timestamp: Time.now }
```
Ref: https://github.com/googleapis/google-cloud-ruby/blob/809ebea224c9b8c9f002c7809afa537b735602d2/google-cloud-spanner/lib/google/cloud/spanner/client.rb#L375

Valid keys for `single_use`
https://github.com/googleapis/google-cloud-ruby/blob/809ebea224c9b8c9f002c7809afa537b735602d2/google-cloud-spanner/lib/google/cloud/spanner/client.rb#L1732-L1736

Reply:
No, that is not required (as is also the default for the other client libraries):
1. Calling `execute_query` without a `single_use` parameter will cause the method to call the `single_use_transaction` method with a nil parameter [here](https://github.com/googleapis/google-cloud-ruby/blob/809ebea224c9b8c9f002c7809afa537b735602d2/google-cloud-spanner/lib/google/cloud/spanner/client.rb#L382).
2. The `single_use_transaction` method will [return nil](https://github.com/googleapis/google-cloud-ruby/blob/809ebea224c9b8c9f002c7809afa537b735602d2/google-cloud-spanner/lib/google/cloud/spanner/client.rb#L1748) when it is called with a nil parameter.
3. That will cause the `session.execute_query` method to be [called with a nil transaction selector](https://github.com/googleapis/google-cloud-ruby/blob/809ebea224c9b8c9f002c7809afa537b735602d2/google-cloud-spanner/lib/google/cloud/spanner/session.rb#L315).
4. This will in the end call the `ExecuteStreamingSql` RPC to be called with an `ExecuteSqlRequest` without a transaction selector. That will cause Cloud Spanner to use the [default transaction selector](https://cloud.google.com/spanner/docs/reference/rpc/google.spanner.v1#google.spanner.v1.ExecuteSqlRequest), which is a single use read-only transaction with a strong timestamp bound.

Reply:
Thanks for information.